core/jobs/batch_jobs/store_profile_images_to_gcs_jobs.py

```diff
@@ -0,0 +1,317 @@
+            'filepath': filepath,
+            'data': webp_binary
+        }
+        return file_dict
+
+    def _make_profile_picture_valid(
+        self, user_model: user_models.UserSettingsModel
+    ) -> user_models.UserSettingsModel:
+        """Generate gravatar for users that have profile picture None.
+
+        Args:
+            user_model: user_models.UserSettingsModel. The user model.
+
+        Returns:
+            user_model: user_models.UserSettingsModel. The updated user model.
+        """
+        if user_model.profile_picture_data_url is None:
+            user_model.profile_picture_data_url = (
+                user_services.fetch_gravatar(user_model.email))
+
+        return user_model
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+        users_with_valid_username = (
+            self.pipeline
+            | 'Get all non-deleted UserSettingsModel' >> ndb_io.GetModels(
+                user_models.UserSettingsModel.get_all(include_deleted=False))
+            | 'Filter valid users with not None username' >> beam.Filter(
+                lambda model: model.username is not None)
+            | 'Make the invalid profile picture valid' >> beam.Map(
+                self._make_profile_picture_valid)
+        )
+
+        write_png_files_to_gcs = (
+            users_with_valid_username
+            | 'Map files for png' >> beam.Map(self._filepath_png)
+            | 'Write png file to GCS' >> gcs_io.WriteFile(mime_type='image/png')
+        )
+
+        total_png_images = (
+            write_png_files_to_gcs
+            | 'Total png images wrote to GCS' >> (
+                job_result_transforms.CountObjectsToJobRunResult(
+                    'TOTAL PNG IMAGES'))
+        )
+
+        write_webp_files_to_gcs = (
+            users_with_valid_username
+            | 'Map files for webp' >> beam.Map(self._filepath_webp)
+            | 'Write webp file to GCS' >> gcs_io.WriteFile(
+                mime_type='image/webp')
+        )
+
+        total_webp_images = (
+            write_webp_files_to_gcs
+            | 'Total webp images wrote to GCS' >> (
+                job_result_transforms.CountObjectsToJobRunResult(
+                    'TOTAL WEBP IMAGES'))
+        )
+
+        return (
+            (
+                total_png_images,
+                total_webp_images
+            )
+            | 'Combine results' >> beam.Flatten()
+        )
+
+
+def png_base64_to_webp_base64(png_base64: str) -> str:
+    """Convert png base64 to webp base64.
+
+    Args:
+        png_base64: str. The png base64 string.
+
+    Returns:
+        str. The webp base64 string.
+    """
+    png_binary = utils.convert_png_data_url_to_binary(png_base64)
+    output = io.BytesIO()
+    image = Image.open(io.BytesIO(png_binary)).convert('RGB')
+    image.save(output, 'webp')
+    webp_binary = output.getvalue()
+    return utils.convert_png_or_webp_binary_to_data_url(
+        webp_binary, 'webp')
```

Comment:
Can you move this inside `AuditProfilePictureFromGCSJob`

Reply:
Done